docs/enhancements/install-sno-using-ibi.md

```diff
@@ -0,0 +1,133 @@
+---
+title: install-sno-using-ibi
+authors:
+  - "@adriengentil"
+creation-date: 2024-03-28
+last-updated: 2024-03-28
+---
+
+# Support Image-Based Installation in Assisted-Installer to install SNO
+
+## Summary
+
+Currently, installing an SNO cluster using the Assisted-Installer takes up to
+~40min using the current flow which installs OCP from the ground-up.
+
+We can reduce the installation time to ~10min by leveraging the Image-Based
+installation method, and as a way to increase the success rate of SNO
+installations as we restore previously working setup.
+
+This enhancement describes the changes to bring to the Assisted-Installer in
+order to perform SNO installation using the Image-Based installation method.
+
+## Motivation
+
+The Image-Based installation brings the following advantages:
+- Users will experience faster SNO installations
+- Increased success rate installations since we restore the state of a
+  previously working SNO setup
+
+### Goals
+
+- Provide an option in Assisted-Installer to allow a user to install a SNO with
+  the Image-Based Installation method
+
+### Non-Goals
+
+- Focus is on Assisted-Installer in SaaS environment, we don’t aim to support
+  other environments at this point
+
+## Proposal
+
+### As a user, I want to install a SNO using the IBI method
+
+#### REST API changes
+
+A new flag will be added to the cluster object [TBD], along with
+extra-parameters specific to IBI [TBD].
+
+`supported-versions` API with be amended to return only the version of OCP
+available for installation with IBI.
+
+#### Feature support
+
+A new feature support will be created for the IBI installation. 
+The feature will be gated by an OCM capability until we make it available to
+all users.
+
+#### Installation
+
+When the user will hit the installation button, the `assisted-installer` will
+restore the image seed on the disk selected for installation:
+- Install RHCOS using `coreos-installer`
+- Partition the disk to create a dedicated `/var/lib/containers`
+- Grow `/` partition
+- Mount `/`, `/boot` and `/var/lib/container`
+- Download and restore the seed on the disk using `lca-cli`
+
+[Reference](https://github.com/openshift-kni/lifecycle-agent/blob/main/ib-cli/installationiso/data/install-rhcos-and-restore-seed.sh)
+
+#### Configuration
+
+After the image seed is restored on the disk, the `assisted-installer` will
+drop the configuration in `/opt/openshift`:
```

Comment:
We'll also need to consider static network configuration.

I think we should be able to continue to provide nmconnection files (as we are currently in assisted) but in "real" IBI we're using nmstate and running nmstatectl on the host so we'll want to make sure the nmconnection path continues to work.

Reply:
Isn't static networking already based on nmstate in assisted (which generate nmconnection files)? or I miss something?
For sure, we need to consider this flow.